### ISSUE TITLE
Synchronization of remote file rename causes exception in social RDBMS storage

### ISSUE THREAD

**Issue by pnedonosko** [exo-addons/cloud-drive-extension#60]
When running in Platform 4.4.0, each remote file rename causes following exception in server logs:

```
2017-02-06 14:32:11,629 | ERROR | Exception on broadcasting events occurs: null [e.k.component.common.ListenerService<clouddrive-thread-8>]
java.lang.reflect.UndeclaredThrowableException: null
	at org.exoplatform.social.core.storage.cache.FutureCache.get(FutureCache.java:164) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage.getSpaceByGroupId(CachedSpaceStorage.java:1305) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.space.impl.SpaceServiceImpl.getSpaceByGroupId(SpaceServiceImpl.java:224) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.wcm.ext.component.activity.listener.Utils.postFileActivity(Utils.java:528) ~[integ-ecms-social-4.4.0.jar:4.4.0]
	at org.exoplatform.wcm.ext.component.activity.listener.FileUpdateActivityListener.onEvent(FileUpdateActivityListener.java:242) ~[integ-ecms-social-4.4.0.jar:4.4.0]
	at org.exoplatform.services.listener.ListenerService.broadcast(ListenerService.java:188) ~[exo.kernel.component.common-2.6.0.jar:2.6.0]
	at org.exoplatform.services.cms.jcrext.activity.EditFilePropertyActivityAction.execute(EditFilePropertyActivityAction.java:67) ~[ecms-core-services-4.4.0.jar:4.4.0]
	at org.exoplatform.services.jcr.impl.ext.action.SessionActionInterceptor.launch(SessionActionInterceptor.java:535) ~[exo.jcr.component.core-1.17.0.jar:1.17.0]
	at org.exoplatform.services.jcr.impl.ext.action.SessionActionInterceptor.postSetProperty(SessionActionInterceptor.java:351) ~[exo.jcr.component.core-1.17.0.jar:1.17.0]
	at org.exoplatform.services.jcr.impl.core.ItemImpl.doUpdateProperty(ItemImpl.java:599) [exo.jcr.component.core-1.17.0.jar:1.17.0]
	at org.exoplatform.services.jcr.impl.core.ItemImpl.doUpdateProperty(ItemImpl.java:413) [exo.jcr.component.core-1.17.0.jar:1.17.0]
	at org.exoplatform.services.jcr.impl.core.NodeImpl.setProperty(NodeImpl.java:2327) [exo.jcr.component.core-1.17.0.jar:1.17.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive.initCommon(JCRLocalCloudDrive.java:5387) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive.initFile(JCRLocalCloudDrive.java:5309) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive.access$3400(JCRLocalGoogleDrive.java:71) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.updateFile(JCRLocalGoogleDrive.java:527) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.syncNext(JCRLocalGoogleDrive.java:342) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.syncFiles(JCRLocalGoogleDrive.java:280) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$SyncCommand.process(JCRLocalCloudDrive.java:1640) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$AbstractCommand.exec(JCRLocalCloudDrive.java:1241) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$CommandCallable.call(JCRLocalCloudDrive.java:868) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$CommandCallable.call(JCRLocalCloudDrive.java:848) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_112]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_112]
Caused by: javax.persistence.PersistenceException: org.hibernate.SessionException: Session is closed!
	at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1387) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1310) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.QueryImpl.getSingleResult(QueryImpl.java:316) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.exoplatform.social.core.jpa.storage.dao.jpa.SpaceDAOImpl.getSpaceByGroupId(SpaceDAOImpl.java:49) ~[social-component-core-jpa-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.jpa.storage.RDBMSSpaceStorageImpl.getSpaceByGroupId(RDBMSSpaceStorageImpl.java:316) ~[social-component-core-jpa-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage$37.execute(CachedSpaceStorage.java:1308) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage$37.execute(CachedSpaceStorage.java:1306) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.loader.CacheLoader.retrieve(CacheLoader.java:40) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.loader.CacheLoader.retrieve(CacheLoader.java:29) ~[social-component-core-4.4.0.jar:4.4.0]
	at org.exoplatform.social.core.storage.cache.FutureCache$1.call(FutureCache.java:116) ~[social-component-core-4.4.0.jar:4.4.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_112]
	at org.exoplatform.social.core.storage.cache.FutureCache.get(FutureCache.java:150) ~[social-component-core-4.4.0.jar:4.4.0]
	... 25 common frames omitted
Caused by: org.hibernate.SessionException: Session is closed!
	at org.hibernate.internal.AbstractSessionImpl.errorIfClosed(AbstractSessionImpl.java:130) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1219) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.internal.QueryImpl.list(QueryImpl.java:101) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.QueryImpl.getSingleResult(QueryImpl.java:287) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	... 34 common frames omitted
2017-02-06 14:32:11,712 | WARN  | Exception reading value for field:  [e.jcr.component.core.TextFieldExtractor<http-nio-8443-exec-1>]
2017-02-06 14:32:11,713 | WARN  | Exception reading value for field:  [e.jcr.component.core.TextFieldExtractor<http-nio-8443-exec-1>]
```

**Comment by pnedonosko** [comment]
Also actual in latest Platform 4.4.1-RC03

```
2017-02-06 14:54:17,411 | INFO  | Google Drive - pnedonosko@exoplatform.com successfully connected. [o.e.clouddrive.rest.ConnectService<clouddrive-thread-1>]
2017-02-06 14:55:14,002 | ERROR | Exception on broadcasting events occurs: null [e.k.component.common.ListenerService<clouddrive-thread-4>]
java.lang.reflect.UndeclaredThrowableException: null
	at org.exoplatform.social.core.storage.cache.FutureCache.get(FutureCache.java:164) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage.getSpaceByGroupId(CachedSpaceStorage.java:1307) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.space.impl.SpaceServiceImpl.getSpaceByGroupId(SpaceServiceImpl.java:224) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.wcm.ext.component.activity.listener.Utils.postFileActivity(Utils.java:528) ~[integ-ecms-social-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.wcm.ext.component.activity.listener.FileUpdateActivityListener.onEvent(FileUpdateActivityListener.java:242) ~[integ-ecms-social-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.services.listener.ListenerService.broadcast(ListenerService.java:188) ~[exo.kernel.component.common-2.6.1-RC03.jar:2.6.1-RC03]
	at org.exoplatform.services.cms.jcrext.activity.EditFilePropertyActivityAction.execute(EditFilePropertyActivityAction.java:67) ~[ecms-core-services-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.services.jcr.impl.ext.action.SessionActionInterceptor.launch(SessionActionInterceptor.java:535) ~[exo.jcr.component.core-1.17.1-RC03.jar:1.17.1-RC03]
	at org.exoplatform.services.jcr.impl.ext.action.SessionActionInterceptor.postSetProperty(SessionActionInterceptor.java:351) ~[exo.jcr.component.core-1.17.1-RC03.jar:1.17.1-RC03]
	at org.exoplatform.services.jcr.impl.core.ItemImpl.doUpdateProperty(ItemImpl.java:599) [exo.jcr.component.core-1.17.1-RC03.jar:1.17.1-RC03]
	at org.exoplatform.services.jcr.impl.core.ItemImpl.doUpdateProperty(ItemImpl.java:413) [exo.jcr.component.core-1.17.1-RC03.jar:1.17.1-RC03]
	at org.exoplatform.services.jcr.impl.core.NodeImpl.setProperty(NodeImpl.java:2327) [exo.jcr.component.core-1.17.1-RC03.jar:1.17.1-RC03]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive.initCommon(JCRLocalCloudDrive.java:5387) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive.initFile(JCRLocalCloudDrive.java:5309) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive.access$3400(JCRLocalGoogleDrive.java:71) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.updateFile(JCRLocalGoogleDrive.java:527) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.syncNext(JCRLocalGoogleDrive.java:342) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.gdrive.JCRLocalGoogleDrive$Sync.syncFiles(JCRLocalGoogleDrive.java:280) [exo-clouddrive-gdrive-services-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$SyncCommand.process(JCRLocalCloudDrive.java:1640) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$AbstractCommand.exec(JCRLocalCloudDrive.java:1241) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$CommandCallable.call(JCRLocalCloudDrive.java:868) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at org.exoplatform.clouddrive.jcr.JCRLocalCloudDrive$CommandCallable.call(JCRLocalCloudDrive.java:848) [exo-clouddrive-services-core-1.5.0.jar:1.5.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_112]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_112]
Caused by: javax.persistence.PersistenceException: org.hibernate.SessionException: Session is closed!
	at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1387) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1310) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.QueryImpl.getSingleResult(QueryImpl.java:316) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	at org.exoplatform.social.core.jpa.storage.dao.jpa.SpaceDAOImpl.getSpaceByGroupId(SpaceDAOImpl.java:49) ~[social-component-core-jpa-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.jpa.storage.RDBMSSpaceStorageImpl.getSpaceByGroupId(RDBMSSpaceStorageImpl.java:316) ~[social-component-core-jpa-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage$37.execute(CachedSpaceStorage.java:1310) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.CachedSpaceStorage$37.execute(CachedSpaceStorage.java:1308) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.loader.CacheLoader.retrieve(CacheLoader.java:40) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.loader.CacheLoader.retrieve(CacheLoader.java:29) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at org.exoplatform.social.core.storage.cache.FutureCache$1.call(FutureCache.java:116) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_112]
	at org.exoplatform.social.core.storage.cache.FutureCache.get(FutureCache.java:150) ~[social-component-core-4.4.1-RC03.jar:4.4.1-RC03]
	... 25 common frames omitted
Caused by: org.hibernate.SessionException: Session is closed!
	at org.hibernate.internal.AbstractSessionImpl.errorIfClosed(AbstractSessionImpl.java:130) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1219) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.internal.QueryImpl.list(QueryImpl.java:101) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
	at org.hibernate.ejb.QueryImpl.getSingleResult(QueryImpl.java:287) ~[hibernate-entitymanager-4.2.21.Final.jar:4.2.21.Final]
	... 34 common frames omitted
2017-02-06 14:55:14,062 | WARN  | Exception reading value for field:  [e.jcr.component.core.TextFieldExtractor<http-nio-8443-exec-2>]
2017-02-06 14:55:14,063 | WARN  | Exception reading value for field:  [e.jcr.component.core.TextFieldExtractor<http-nio-8443-exec-2>]
```

**Comment by pnedonosko** [comment]
Should be fixed in 1.5.4 and 1.6.0.
